### PR TITLE
Add documentation about incr. comp. query attributes and the "projection query" pattern.

### PR DESCRIPTION
**Pull request by michaelwoerister**

### CODE REVIEW

File: src/queries/incremental-compilation-in-detail.md [hunk 1]
```diff
@@ -401,6 +401,95 @@ deal with all of the above but so far that seemed like more trouble than it
...
+    - Not computing the fingerprint can save quite a bit of time because
+      fingerprinting is expensive, especially for large, complex values.
+
+    - Without the fingerprint, the system has to unconditionally assume that
+      the result of the query has changed. As a consequence anything depending
+      on a `no_hash` query will always be re-executed.
+
+ - `cache_on_disk_if` - This attribute is what determines which query results
+   are persisted in the incremental compilation query result cache. The
+   attribute takes an expression that allows to make per query invocation
```
Comment (mark-i-m) [inline src/queries/incremental-compilation-in-detail.md]:
```suggestion
   attribute takes an expression that allows per query invocation
```

**Comment by michaelwoerister** [comment]
I applied the suggested changes. I don't really know what the build failures are about since they are in links that I didn't touch.

**Comment by mark-i-m** [comment]
@michaelwoerister Thanks! Could you rebase on top of master (which has a bunch of link fixes)?

**Comment by mark-i-m** [comment]
Ping @michaelwoerister Could you rebase please? r=me after that

File: src/queries/incremental-compilation-in-detail.md [hunk 2]
```diff
@@ -401,6 +401,97 @@ deal with all of the above but so far that seemed like more trouble than it
...
+modifiers affect certain aspects of how the system treats the query with
+respect to incremental compilation:
+
+ - `eval_always` - A query with the `eval_always` attribute is re-executed
+   unconditionally during incremental compilation. I.e. the system will not
+   even try to mark the query's dep-node as green. This attribute has two use
+   cases:
+
+    - Queries that are **inputs** to the system must be marked as `eval_always`
+      since we have no other way of checking whether their value has changed.
```
Comment (Zoxc) [inline src/queries/incremental-compilation-in-detail.md:417]: I think it's more accurate to phrase this as `eval_always` queries can read inputs (from files, global state, etc). They can also produce side effects like changing global state and writing to files.

**Comment by JohnTitor** [comment]
Rebased and applied suggestions from @Zoxc, r? @mark-i-m

**Pull request merged**